eclair-core/src/main/scala/fr/acinq/eclair/wire/TlvCodecs.scala

```diff
@@ -104,14 +105,17 @@ object TlvCodecs {
   val ltu16: Codec[Int] = variableSizeBytes(uint8, tu16)
 
   private def validateGenericTlv(g: GenericTlv): Attempt[GenericTlv] = {
-    if (g.tag.toBigInt % 2 == 0) {
+    if (g.tag < TLV_TYPE_HIGH_RANGE && g.tag.toBigInt % 2 == 0) {
       Attempt.Failure(Err("unknown even tlv type"))
     } else {
       Attempt.Successful(g)
     }
   }
 
-  private val genericTlv: Codec[GenericTlv] = (("tag" | varint) :: variableSizeBytesLong(varintoverflow, bytes)).as[GenericTlv].exmap(validateGenericTlv, validateGenericTlv)
+  private val genericTlv: Codec[GenericTlv] = (("tag" | varint) :: variableSizeBytesLong(varintoverflow, bytes)).as[GenericTlv].exmap(
```

Comment:
Now that we chose to allow everything in the range > 2^16, I think we should keep the stricter version for tlv < 2^16.
So I'd keep the previous code here, WDYT?

Reply:
Agreed. It was a workaround  to play nice with plugins but since we now allow encoding/decoding even types above the high range it feels correct to *disallow* encoding them if they're not in the high range. Changes for this are in 8e5d22b